server/isomorphic-routing/index.js

```diff
@@ -6,22 +6,32 @@ import { setSection as setSectionMiddlewareFactory } from '../../client/controll
 
 export function serverRouter( expressApp, setUpRoute, section ) {
 	return function( route, ...middlewares ) {
-		expressApp.get(
-			route,
-			setUpRoute,
-			combineMiddlewares(
-				setSectionMiddlewareFactory( section ),
-				...middlewares
-			),
-			serverRender
-		);
+		if ( middlewares.length === 0 && typeof route === 'function' && route.length === 3 ) {
```

Comment:
> app.get( someMw ) is a shorthand for app.get( '*', someMw )

I'm not certain about that https://expressjs.com/en/api.html#app.get.method. It should be fine to have these checks anyway, but we need to document it.